apps/test/unit/code-studio/pd/application/TeacherApplicationTest.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {expect} from 'chai';
+import {mount, shallow} from 'enzyme';
+import sinon from 'sinon';
+import {isolateComponent} from 'isolate-components';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+import {PageLabels} from '@cdo/apps/generated/pd/teacherApplicationConstants';
+import TeacherApplication from '@cdo/apps/code-studio/pd/application/teacher/TeacherApplication';
+
+describe('TeacherApplication', () => {
+  const fakeOptionKeys = Object.values(PageLabels).reduce(
+    (acc, cur) => acc.concat(Object.keys(cur)),
+    []
+  );
+  const fakeOptions = fakeOptionKeys.reduce(
+    (acc, cur) => ({...acc, [cur]: ['1', '2', '3']}),
+    {}
+  );
+  const defaultProps = {
+    apiEndpoint: '/path/to/endpoint',
+    requiredFields: [],
+    options: fakeOptions,
+    accountEmail: 'user@email.com',
+    userId: 1
+  };
+
+  let ga;
+
+  beforeEach(() => {
+    ga = sinon.fake();
+    window.ga = ga;
+    window.sessionStorage.removeItem('TeacherApplication');
+  });
+
+  afterEach(() => {
+    window.ga = undefined;
+    sinon.restore();
+    window.sessionStorage.removeItem('TeacherApplication');
+  });
+
+  it('Does not set schoolId if not provided', () => {
+    const page = mount(<TeacherApplication {...defaultProps} />);
```

Comment:
Open to trying to refactor these tests ... wasn't sure the best testing strategy here and didn't want the PR to get too big (these tests were originally in `TeacherApplicationComponentTest`

Reply:
for some reason github isn't showing the file move, so these all look like additions in the github UI. if there is a substantial amount of new test cases here that you want eyes on, it would be great if you could call those out. otherwise, if this is just a move, I think it is fine to leave these as they were in the original file.

Reply:
Yes very weird the other file move is shown but this one isn't ... I'll flag the new tests (they're all together below)